examples/bits.rs

```diff
@@ -1,98 +1,207 @@
-use std::fmt;
+//! View bits in memory with turtles
+//!
+//! This example uses [@myrrlyn]'s [`bitvec`] crate to turn data into strings of
+//! bits, and then draws them on the screen.
+//!
+//! You are encouraged to change both the data used to seed the turtle, and the
+//! `bitvec` calls that control how the turtle acts, to see what changes.
+//!
+//! [@myrrlyn]: //github.com/myrrlyn
+//! [`bitvec`]: //crates.io/crates/bitvec
+
+// The `bitvec` crate provides a lot of types; we only need a few:
+use bitvec::prelude::*;
+use turtle::Turtle;
+
+// Modify these constants to change the behavior of the example.
+
+/// This controls the width of the drawn line for each bit.
+const BIT_WIDTH: f64 = 20.0;
+
+/// This controls the vertical spacing between rows of bit lines.
+const BIT_HEIGHT: f64 = 10.0;
+
+/// This text will be inspected as individual bytes, and drawn on the screen.
+/// You can change it to see what different text looks like when viewed as bits.
+///
+/// The example program will print more information about the parts of the text
+/// to the console while the turtle draws, so that you can see how each glyph
+/// corresponds to parts of the rendered memory.
+static TEXT: &str = "¡Hola, mundo! 🌍🌏🌎";
+
+/// This number will have its bit pattern printed on screen. Rust provides some
+/// interesting numbers in its standard library; you can replace this with other
+/// numbers to see what they look like. Pi is provided as the default solely
+/// because it is well-known, and has an interesting pattern.
+const NUMBER: f64 = std::f64::consts::PI;
+
+// These are helpers for the turtle's actions.
+
+/// Set the horizontal spacing between successive bits in a row
+const BIT_MARGIN: f64 = BIT_WIDTH / 2.0;
+/// Compute the total width of a bit plus its spacing
+const BIT_BOX: f64 = BIT_WIDTH + BIT_MARGIN;
 
 fn main() {
-    let mut bits = BitVec::with_capacity(17);
-    bits.set(0, true);
-    bits.set(1, true);
-    bits.set(3, true);
-    bits.set(8, true);
-    bits.set(10, true);
-    bits.set(16, true);
-    println!("{}", bits);
-    bits.set(0, false);
-    bits.set(8, false);
-    println!("{}", bits);
-}
+    // This block sets up the turtle to draw bits more or less centered in the
```

Comment:
What do you think of splitting `main` into two functions? Feel free to choose good names based on what they do. I was thinking something like `draw_text_bytes` and `draw_number_bytes` might make sense. I'll leave that to your judgment.

Reply:
Done